tests/Unit/CacheInvalidatorTest.php

```diff
@@ -184,8 +184,29 @@ public function testMethodException()
      */
     public function testProxyClientExceptionsAreLogged()
     {
-        $unreachableException = ProxyUnreachableException::proxyUnreachable('http://127.0.0.1', 'Couldn\'t connect to host');
-        $responseException    = ProxyResponseException::proxyResponse('http://127.0.0.1', 403, 'Forbidden');
+        $failedRequest = \Mockery::mock('\Psr\Http\Message\RequestInterface')
+            ->shouldReceive('getHeaderLine')->with('Host')->andReturn('127.0.0.1')
+            ->getMock();
+        $adapterException = \Mockery::mock(
+            '\Http\Adapter\Exception\HttpAdapterException',
+            ['Couldn\'t connect to host']
+        )
+            ->shouldReceive('getRequest')->andReturn($failedRequest)
+            ->getMock();
+        ;
+        $unreachableException = ProxyUnreachableException::proxyUnreachable($adapterException);
+        
+        $response = \Mockery::mock('\Psr\Http\Message\ResponseInterface')
+            ->shouldReceive('getStatusCode')->andReturn(403)
+            ->shouldReceive('getReasonPhrase')->andReturn('Forbidden')
+            ->getMock();
+        $adapterException = \Mockery::mock('\Http\Adapter\Exception\HttpAdapterException')
+            ->shouldReceive('getMessage')->andReturn('Forbidden')
+            ->shouldReceive('getRequest')->andReturn($failedRequest)
+            ->shouldReceive('getResponse')->andReturn($response)
+            ->getMock()
+        ;
+        $responseException = ProxyResponseException::proxyResponse($adapterException);
```

Comment:
that sounds interesting!
i guess we do not need to fix this before merging, but can replace it
once its ready. some MessageInterface to json / from json would be very
nice for general purpose. maybe a testing library together with the mock
adapter?

php-vcr is quite cool, but operating on a low level. we use it in a
project with lots of functional tests about things talking to a REST
api. but here it would be overkill. and its only for functional testing,
not in the spirit of unit testing i think.